### DISCUSSION TITLE
Get stories and understand their logic

### DISCUSSION THREAD

**Discussion by biox86** [sbertix/Swiftagram#240] [Q&A] [answered]
Hello, I started to make a few tests on the story endpoints and I was wondering if they are "complicated" as they seems to be, or if I'm just missing something.

I'm trying to get the stories of people I'm following as they appear in the Instagram app, so I'm calling _Endpoint.recent.stories_.
It returns a _TrayItem.Collection_ that is ordered in the same exact way as the Instagram app ( so that's seems to be the right one ).

Every item can have its own _items_ parameter ( basically an array of Media objects ) that seems to be just the unwatched stories.
On the other hand, every item also has a _mediaIds_ parameter in its Wrapped object that seems to contain all the media ids of the available stories ( already watched or not ).

At the moment I'm interested in the second one because I'm trying to display all the available stories.

With those ids I can now call the _Endpoint.media(#media_id)_ endpoint to get the real story medias.
This call returns a _Media.Unit_ object.
This object has a _media_ parameter, but it seems to always be nil.
This object also has an _items_ parameter ( it seems an array of medias ) in its Wrapped object that seems to contain the media I'm searching for ( the video/image of the story itself ).

Now, do I really have to do this nested endpoint calls and work on the Wrappers to get all the available story medias or did I just missed some endpoint call or parameter?

I hope to have written in an understandable way, If I haven't been clear ( or grammatically correct ) enough, just let me know and I'll try my best to explain myself better.

...and thanks in advance for any clarification or confirmation of the process to get the stories!

**Comment by sbertix** [comment, marked as answer]
Hey @biox86, 

yep, you're definitely overcomplicating things 😊 
All you need to do is check for available stories with `Endpoint.recent.stories` and then load them with `Endpoint.user(_:).stories` one by one, as you actually want to load them, passing the `TrayElement` `identifier` as the `Endpoint.user(_:).stories` argument.

**Do not** rely on `Endpoint.users(_:).stories` with all `identifier`s at once as it only loads media for the first one (just like `Endpoint.recent.stories`). Again: `Endpoint.user(_:).stories` one at a time, not `Endpoint.users(_:).stories`.

**Reply by biox86** [reply]
It really felt too much complicated, glad I asked!

I just tried using the `Endpoint.user(_:).stories` passing the `TrayElement` `identifier` and It seems to work as expected.

It returns a `TrayItem.Unit`, from its `item` parameter I can get the `TrayItem` object and from its `items` parameter I can get the `[Medias]` array with all the available stories' medias.

Thanks a lot for your crazy fast help as usual!!

**Reply by olegjunios** [reply]
Hey!
I am trying to get stories using Swiftagram
but can't get ReceiveValue
could you help me please?

**Comment by olegjunios** [comment]
Hey!
I am trying to get stories using Swiftagram
but can't get ReceiveValue
could you help me please?
